### PR TITLE
Started Play 2.5 streams migration docs

### PR DESCRIPTION
**Pull request by richdougherty**
I split out the streams and Java migration docs into separate files because they're each quite big on their own. The streams migration docs are still being worked on by me, but I'm pushing them now so that other people can work on the other parts of the docs.

### CODE REVIEW

File: documentation/build [hunk 1]
```diff
@@ -1,3 +1,6 @@
 #! /bin/sh
 
+DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+
+cd $DIR
```
Comment (richdougherty) [inline documentation/build]: So the *documentation/build* script can be called from outside the directory.

File: documentation/manual/releases/release25/Highlights25.md [hunk 2]
```diff
@@ -30,6 +30,10 @@ This page highlights the new features of Play 2.5. If you want learn about the c
 
 - link to information in migration docs
 
+### Better control over WebSocket frames
+
+The Play 2.5 WebSocket API gives you direct control over WebSocket frames. You can now send and receives binary, text, ping, pong and close frames. If you don't want to worry about this level of detail, Play will still automatically convert your JSON or XML data into the right kind of frame.
```
Comment (richdougherty) [inline documentation/manual/releases/release25/Highlights25.md]: I think this is a great new feature.

File: documentation/manual/releases/release25/migration25/JavaMigration25.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
+# Java Migration Guide
```
Comment (richdougherty) [inline documentation/manual/releases/release25/migration25/JavaMigration25.md]: This file's content was moved unchanged from `Migration25.md`.

File: framework/src/play/src/main/scala/play/api/mvc/Results.scala [hunk 4]
```diff
@@ -374,7 +374,13 @@ trait Results {
     private def streamFile(file: Source[ByteString, _], name: String, length: Long, inline: Boolean): Result = {
       Result(
         ResponseHeader(status,
-          Map(CONTENT_DISPOSITION -> s"""${if (inline) "inline" else "attachment"}; filename="$name"""")),
+          Map(
+            CONTENT_DISPOSITION -> {
+              val dispositionType = if (inline) "inline" else "attachment"
+              dispositionType + "; filename=\"" + name + "\""
```
Comment (richdougherty) [inline framework/src/play/src/main/scala/play/api/mvc/Results.scala]: This change is simply to work around a bug in Sublime Text's syntax highlighting. :)

File: documentation/build (hunk 1 above)
Comment (jroper) [inline documentation/build]: I'm deleting this script anyway.

File: framework/src/play/src/main/scala/play/api/mvc/Results.scala (hunk 4 above)
Comment (jroper) [inline framework/src/play/src/main/scala/play/api/mvc/Results.scala]: if statements shouldn't be embedded in string interpolation, that's asking for trouble.

**Comment by jroper** [comment]
Build failure was either a transient infrastructure failure, or perhaps a corrupt cache - restarted.

**Pull request merged**